contrib/packaging_python/Readme.md

```diff
@@ -0,0 +1,123 @@
+
+The main script for building and uploading wheels is implemented in the
+`package-python-wheel` action. It builds a Docker container that is a slightly
+extended version of `manylinux2014` (see `manylinux2014/Dockerfile`), runs the `mk_clean_wheel.sh` script for every supported python version and finally collects and uploads the wheels.
+
+The `mk_clean_wheel.sh`:
+
+1. prepares the environment by creating and activating a proper python venv and installing some packages;
+2. configures cvc5 appropriately using the `mk_build_dir.py` script, which makes sure that the venv python version is used;
+3. builds cvc5 and the python extension via the `mk_wheel.py` script;
+4. postprocesses the wheel (with `auditwheel`) and moves it out of the build folder.
+
+
+## Uploading to PyPi
+
+To upload a wheel to test PyPi,
+
+    twine upload --repository testpypi -u $USERNAME -p $PASSWORD PATH_TO_WHEEL
+
+Note that you will need a TestPyPi login. Once it has been uploaded, you can
+test (from anywhere, not just the container) that the wheel works by installing
+it from the TestPyPi repository.
+
+```
+python3 -m pip install --index-url https://test.pypi.org/simple/ pycvc5==<cvc5 version number>
+python3 -c "import pycvc5; solver = pycvc5.Solver(); print(solver.getIntegerSort())"
+```
+
+You can remove the repository argument from the upload command to upload to
+real PyPi.
+
+## What goes on within the Docker container
+
+### Setup
+
+Compared to `manylinux2014`, the container comes with `ccache` and some cvc5
+dependencies preinstalled. Additionally, we point `cmake` to `ccache` using the `CMAKE_<LANG>_COMPILER_LAUNCHER` environment variables.
+
+The idea is to mount a cvc5 checkout into the Docker container at `/home/pycvc5`.
+This way, we don't need to care about cloning cvc5 within Docker and the `ccache` cache is automatically stored persistently outside of the container.
+
+### Building
+
+To build the wheel for a specific python version, simply run `mk_clean_wheel.sh` within the docker container, and pass it the python binary it shall use (`/opt/python/cp<something>/bin/python`) and options to the cvc5 configure script (usually something like `production --auto-download`).
+The script will create a new build folder (`build_wheel`), configure and build cvc5 accordingly and store the wheel within a `wheel-<version>` folder.
+
+### Testing
+
+From within Docker, you can test a wheel as follows: to ensure that the `cvc5`
+library is linked correctly (and not looking at a local path). To do this,
+start from the top-level `cvc5` directory and run the following:
+```
+# delete build directory to get rid of local library
+rm -r build
+# start virtualenv
+source ./env<Python version>/bin/activate
+# install the wheel
+pip install <path to wheel for this Python version>
+# run a small example to make sure it works
+python3 -c "import pycvc5; solver = pycvc5.Solver(); print(solver.getIntegerSort())"
+```
+
+### Test PyPi (in VM)
```

Comment:
```suggestion
### Test PyPi (in Docker)
```

Reply:
We don't actually do this from within the container but from the CI...